src/maxdiffusion/metrics/clip/clip_encoder.py

```diff
@@ -52,12 +56,43 @@ class CLIPEncoderFlax:
     def __init__(self, pretrained="laion/CLIP-ViT-H-14-laion2B-s32B-b79K"):
         assert pretrained is not None
 
-        self.model = jax.jit(FlaxCLIPModel.from_pretrained(pretrained))
+        self.model = FlaxCLIPModel.from_pretrained(pretrained)
+        self.logit_scale = jnp.exp(self.model.params['logit_scale'])
+        self.get_image_features = jax.jit(self.model.get_image_features)
+        self.get_text_features = jax.jit(self.model.get_text_features)
+        self.model = jax.jit(self.model)
         self.processor = AutoProcessor.from_pretrained(pretrained)
-    
+        
     def get_clip_score(self, text, image):
+        # See https://github.com/huggingface/transformers/blob/8c12690cecbb97e187861e386f7a0ac790e4236c/src/transformers/models/clip/modeling_flax_clip.py#L1214
+        inputs = self.processor(images=image, return_tensors="jax")
+        image_embeddings = self.get_image_features(**inputs)
+        inputs = self.processor(text=text, return_tensors="jax", padding="max_length", truncation=True)
+        text_embeddings = self.get_text_features(**inputs)
 
-        inputs = self.processor(text=text, images=image, return_tensors="jax", padding="max_length", truncation=True)
-        outputs = self.model(**inputs)
+        return CLIPEncoderFlax._calculate_clip_score(text_embeddings, image_embeddings, self.logit_scale)
 
-        return outputs.logits_per_image / 100
+    @staticmethod
+    @jax.jit
+    def _calculate_clip_score(text_embeddings, image_embeddings, logit_scale):
+        image_embeddings /= jnp.linalg.norm(image_embeddings, axis=-1, keepdims=True)
+        text_embeddings /= jnp.linalg.norm(text_embeddings, axis=-1, keepdims=True)
+
+        image_embeddings = jnp.expand_dims(image_embeddings, 1)
```

Comment:
Frustratingly, there seems to be a precision issue between Jax matmul vs jax sum and element-wise multiplication. If I went with the latter, with simpler code not requiring this batch_matmul, the difference in scores between PyTorch implementation above and Jax sometimes exceeded 1e-3. Assuming we need to keep within that margin (and that the pytorch implementation we had was the ground truth), I switched to this.